.github/workflows/ngsa-pr.yaml

```diff
@@ -27,16 +27,45 @@ jobs:
         registry: ghcr.io
         username: ${{ secrets.GHCR_ID }}
         password: ${{ secrets.GHCR_PAT }}
-
+    - name: Set Secrets
+      run: |
+            echo -n ${{ secrets.NGSA_PRE_COSMOS_RO_KEY}} >| secrets/CosmosKey
+            echo -n ${{ secrets.NGSA_PRE_COSMOS_URL }} >| secrets/CosmosUrl
     - name: Docker Build
       run: |
-        docker build . -t image
-        
+        docker build . --progress auto -t image
+
+    - name: Docker Run and Test
+      run: |
+            #docker run --rm image --version
+            #docker network create runandtestnet 
+            #docker run --rm --net=runandtestnet --network-alias=ngsacsapi --name ngsacs -p 8080:8080 image
+            
+            docker run --rm -d --name ngsacs -p 8080:8080 image
+
+            echo "Waiting for web server to start ..."
+            sleep 25
+
+            #docker run --rm --net=runandtestnet ghcr.io/retaildevcrews/ngsa-lr:beta -l 500 -s http://ngsacsapi:8080/ --max-errors 1 -f baseline.json 
+
```

Comment:
why not use WebV?

Reply:
After a discussion with Joseph he explained that WebV would be easier and faster to run, and it is a global dotnet tool, however we would be limited to add specific capabilities to it if we need to, on the other hand Loderunner might be better long term if we find ourselves doing so in the feature.